File: shlomif-project-euler/47/brian.py
#!/usr/bin/env python

from six.moves import range

Limit = 1000000         # Search under 1 million for now
factors = [0] * Limit   # number of prime factors.
count = 0
for i in range(2, Limit):
    if factors[i] == 0:
        # i is prime
        count = 0
        val = i
        while val < Limit:
            factors[val] += 1
            val += i
    elif factors[i] == 4:
        count += 1
        if count == 4:
            print(i-3)  # First number
            break
    else:
        count = 0

File: shlomif-project-euler/47/brian.pl
use strict;
use warnings;

my $Limit   = 1_000_000;
my @factors = ( (0) x $Limit );
my $count   = 0;
for my $i ( 2 .. $Limit )
{
    if ( $factors[$i] == 0 )
    {
        # i is prime.
        $count = 0;
        my $val = $i;
        while ( $val < $Limit )
        {
            $factors[$val]++;
            $val += $i;
        }
    }
    elsif ( $factors[$i] == 4 )
    {
        if ( ++$count == 4 )
        {
            print "Found " . ( $i - 3 ) . "\n";
            exit;
        }
    }
    else
    {
        $count = 0;
    }
}
